Refactor detectLinks

andyklimczak opened the pull request:
This replaces the link detection with [autolink-java](https://github.com/robinst/autolink-java#url-extraction) to detect the links in post text.

Done to fix #278 primarily, but will cover a lot more cases now too

Not sure how you feel about adding dependencies to Clover (autolink-java has no dependencies itself)

Floens commented:
Thanks, this is a good addition to the hacky linker we have now.
I try to keep the dependencies down as much as possible. This library is small and does the job well (and only does this job) so I think it's all good.

Quick question before I merge, does this work when the lines start with `>`? For example does `>https://example.com` resolve?

And lastly, the license of the library needs to be added to the licenses.html file, located in the assets/html folder.

andyklimczak commented:
yes, it covers the ```>``` case:
![screenshot_2017-03-21-11-59-32](https://cloud.githubusercontent.com/assets/4869742/24157109/d57ccfc2-0e2e-11e7-93a9-5c23eff82a29.png)

I also added tests for our use case to robinst/autolink-java#11

and I added the license

Floens commented:
You've outdone yourself with the tests, merging, thanks!
